apps/src/applab/applab.js

```diff
@@ -1154,9 +1240,9 @@ studioApp.reset = function(first) {
   divApplab.parentNode.replaceChild(newDivApplab, divApplab);
 
   divApplab = document.getElementById('divApplab');
-  if (Applab.levelHtml) {
-    divApplab.innerHTML = Applab.levelHtml;
-  }
+  var isDesignMode = window.$ && $('#codeModeButton').is(':visible');
+  Applab.parseFromLevelHtml(divApplab, isDesignMode);
+
   divApplab.addEventListener('click', Applab.onDivApplabClick);
```

Comment:
This section (LL 1232-1244) seems a little more convoluted than necessary.  It:

1. Finds divAppLab
2. Empties divAppLab
3. Clones divAppLab
4. Swaps the clone out for the existing divAppLab
5. Finds divAppLab again (isn't newDivAppLab still a handle on this?)
6. Populates divAppLab (the new one now) from the level HTML

How close can we get to something like this?
```javascript
// Make new root
var newDivApplab = document.createElement('div');
// Fill new root
Applab.parseFromLevelHtml(newDivApplab, this.isDesignMode_());
// Swap in new root
var oldDivApplab = document.getElementById('divApplab');
oldDivApplab.parentNode.replaceChild(newDivApplab, oldDivApplab);
```

I guess my two questions are: Why do we have to clone the original root element, and why don't we put the swap at the end?